tests/parser/statements2.py

```diff
@@ -0,0 +1,10 @@
+a and b
+a and b or c
+a or b
+a or b and c
+
+(a and b and c or (x or (y and (z or i)) or j))
+(a and (b and (c or d)) and (e and f) or x) and y
+
+# # TODO: Make this work similar to Old Parser
+# (a or b and c) or z
```

Comment:
```py
(a or b) or c
```
The problem here is that, as parenthesis information is not used or stored by the macros, it is difficult to handle the above example:
OLD Parser AST:
```console
$ lpython --show-ast examples/expr2.py --tree
└-Module
  |-body=↧
  | └-Expr
  |   └-value=BoolOp
  |     |-boolopType=Or
  |     └-values=↧
  |       |-BoolOp
  |       | |-boolopType=Or
  |       | └-values=↧
  |       |   |-Name
  |       |   | |-id=a
  |       |   | └-expr_contextType=Load
  |       |   └-Name
  |       |     |-id=b
  |       |     └-expr_contextType=Load
  |       └-Name
  |         |-id=c
  |         └-expr_contextType=Load
  └-type_ignores=↧
```
NEW Parser AST (this PR):
```console
$ lpython --show-ast examples/expr2.py --tree --new-parser
└-Module
  |-body=↧
  | └-Expr
  |   └-value=BoolOp
  |     |-boolopType=Or
  |     └-values=↧
  |       |-Name
  |       | |-id=a
  |       | └-expr_contextType=Load
  |       |-Name
  |       | |-id=b
  |       | └-expr_contextType=Load
  |       └-Name
  |         |-id=c
  |         └-expr_contextType=Load
  └-type_ignores=↧
```